sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlockBlobTest.java

```diff
@@ -0,0 +1,427 @@
+
+@RunWith(DataProviderRunner.class)
+@Ignore
+public class BlockBlobTest {
+    private String containerName;
+    private String blobName;
+    private static StorageBlobAsyncClient asyncClient;
+    private static StorageBlobClient syncClient;
+
+    @DataProvider
+    public static Object[][] headers() throws NoSuchAlgorithmException {
+        return new Object[][] {
+            {null,      null,          null,       null,       null,                                                       null},    // 0
+            {"control", "disposition", "encoding", "language", MessageDigest.getInstance("MD5").digest(getDefaultData()), "type"},   // 1
+        };
+    }
+
+    // TODO: (gapra) Add iftags
+    @DataProvider
+    public static Object[][] accessConditionsSuccess() {
+        return new Object[][] {
+            {null,    null,    null,         null},       // 0
+            {oldDate, null,    null,         null},       // 1
+            {null,    newDate, null,         null},       // 2
+            {null,    null,    receivedEtag, null},       // 3
+            {null,    null,    null,         garbageEtag} // 4
+        };
+    }
+
+    // TODO: (gapra) Add iftags
+    @DataProvider
+    public static Object[][] accessConditionsFail() {
+        return new Object[][] {
+            {newDate, null,    null,        null},        // 0
+            {null,    oldDate, null,        null},        // 1
+            {null,    null,    garbageEtag, null},        // 2
+            {null,    null,    null,        receivedEtag} // 3
+        };
+    }
+
+    @BeforeClass
+    public static void setupClass() {
+        asyncClient = initializeDefaultAsyncBlobClientBuilder(enableFiddler()).build();
+        syncClient = initializeDefaultSyncBlobClientBuilder(enableFiddler()).build();
+    }
+
+    @Before
+    public void setupTest() {
+        // Create container
+        containerName = generateResourceName();
+        syncClient.createContainer(containerName);
+
+        // Create blob
+        blobName = generateResourceName();
+        String blockId = generateBlockID();
+        syncClient.stageBlock(containerName, blobName, blockId, getDefaultData(), null);
+        List<String> blockIds = new ArrayList<>();
+        blockIds.add(blockId);
+        syncClient.commitBlockList(containerName, blobName, blockIds, false);
+    }
+
+    @After
+    public void teardownTest() {
+        syncClient.deleteContainer(containerName);
+    }
+
+    @Test
+    public void stageBlock() {
+        // Setup
+        String blockId = generateBlockID();
+
+        // When
+        BlockBlobsStageBlockResponse response = syncClient.stageBlockWithRestResponse(containerName, blobName, blockId, getDefaultData(), null, null, null ,null ,null ,null, null);
+
+        // Then
+        assertEquals(201, response.getStatusCode());
+        BlockBlobStageBlockHeaders headers = response.getDeserializedHeaders();
+        assertNotNull(headers.getXMsContentCrc64());
```

Comment:
I think we should change the name of this header or possibly remove it

Reply:
We can handwrite public types and then this will go away. I will add this to our list

Reply:
Added a comment in another test that also referenced this getter